### ISSUE TITLE
Output of the tool on `bevy-audio` seems like it's reporting a false positive 

### ISSUE THREAD

**Issue by tonowak** [obi1kenobi/cargo-semver-checks#274]
### Steps to reproduce the bug with the above code

```
git clone git@github.com:bevyengine/bevy.git
cd bevy/crates/bevy_audio
git checkout 1efdbb7e # for consistent output
cargo run --manifest-path=../../../cargo-semver-check/Cargo.toml -- semver-checks check-release
```
Probably `cargo semver-check check-release` instead of `cargo run ...` is enough, but I cannot confirm it because of https://github.com/obi1kenobi/cargo-semver-checks/issues/272

### Actual Behaviour

```
   Compiling cargo-semver-checks v0.15.2 (/home/tonowak/zpp/cargo-semver-check)
    Finished dev [unoptimized + debuginfo] target(s) in 5.34s
     Running `/home/tonowak/zpp/cargo-semver-check/target/debug/cargo-semver-checks semver-checks check-release`
    Updating index
     Parsing bevy_audio v0.9.0 (current)
     Parsing bevy_audio v0.9.0 (baseline)
    Checking bevy_audio v0.9.0 -> v0.9.0 (no change)
   Completed [   0.824s] 33 checks; 31 passed, 2 failed, 0 unnecessary

--- failure trait_unsafe_added: pub trait became unsafe ---

Description:
A publicly-visible trait became `unsafe`, so implementing it now requires an `unsafe impl` block.
        ref: https://doc.rust-lang.org/book/ch19-01-unsafe-rust.html#implementing-an-unsafe-trait
       impl: https://github.com/obi1kenobi/cargo-semver-check/tree/v0.15.2/src/queries/trait_unsafe_added.ron

Failed in:
  trait bevy_audio::Sample in file :

--- failure trait_unsafe_removed: pub unsafe trait became safe ---

Description:
A formerly-unsafe public trait became safe, so it is no longer allowed to be implemented with an `unsafe impl` block.
        ref: https://doc.rust-lang.org/book/ch19-01-unsafe-rust.html#implementing-an-unsafe-trait
       impl: https://github.com/obi1kenobi/cargo-semver-check/tree/v0.15.2/src/queries/trait_unsafe_removed.ron

Failed in:
  trait bevy_audio::Sample in file :
       Final [   0.844s] semver requires new major version: 2 major and 0 minor checks failed
```

### Expected Behaviour

idk yet. See Additional Context.

#### Software version

cargo-semver-checks 0.15.2 (f6b41f4-modified)

#### Operating system

Linux 6.1.1-arch1-1

#### Command-line

```bash
target/debug/cargo-semver-checks semver-checks --bugreport
```

#### cargo version

```
> cargo -V
cargo 1.66.0 (d65d197ad 2022-11-15)
```

#### Compile time information

- Profile: debug
- Target triple: x86_64-unknown-linux-gnu
- Family: unix
- OS: linux
- Architecture: x86_64
- Pointer width: 64
- Endian: little
- CPU features: fxsr,sse,sse2
- Host: x86_64-unknown-linux-gnu

### Additional Context

The output is weird -- it doesn't have a `span` and the `trait bevy_audio::Sample` has both become unsafe and safe after being both unsafe and safe (?).

It really seems like a false-positive. This PR is a tracking issue for inspecting the tool's output and `bevy-audio` rustdoc/sourcecode.
